build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/GitCommitInfo.cs

```diff
@@ -0,0 +1,322 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using IOFile = System.IO.File;
+
+namespace Xamarin.Android.BuildTools.PrepTasks
+{
+	public sealed class GitCommitInfo : Task
```

Comment:
This should instead inherit from the `<Git/>` task, in the same directory.  See also e.g. `<GitCommitHash/>`, `<GitBranch/>`, etc.

Reply:
There should probably be a comment mentioning that this class needs to be kept in sync with `get-git-branch.sh`.

Reply:
@grendello doesn't want to deal with the "state machine" shenanigans necessary to use `ToolTask` with multiple process invocations.  Therefore, `Task` as a base class is appropriate.

Reply:
No, that was a conscious decision. Git task derives from ToolTask which is inherently dumb to use when you need to invoke a command more than once - you need to create a state machine so that you can return different parameters from `GenerateCommandLineCommands` and then call `base.Execute()`, then use the state machine again in `LogEventsFromTextOutput` to place the results where they should be. With code like in `GitCommitInfo` it would quickly become messy.